### ISSUE TITLE
[BUG] Modal with header and footer contains confusing landmarks

### ISSUE THREAD

**Issue by RasmusKjeldgaard** [kirbydesign/designsystem#3921]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
The [ARIA guiding principles](https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/general-principles.html) state the following:

> Note that wrapping the content of a modal dialog in a landmark region is unnecessary. A landmark that wraps modal content cannot provide any benefit to users because it is not perceivable unless the modal is open. In addition, when a modal is open, a landmark containing its content is superfluous because the modal itself is a container that provides both a name and boundaries. 

The `ion-footer` that we use inside the `kirby-modal-footer` has  `role="contentinfo"`.

`contentinfo` on MDN is described as: 

> The contentinfo role defines a footer, containing identifying information such as copyright information, navigation links, and privacy statements, found on every document within a site. This section is commonly called a footer.

Unlike a page footer, this seems like a bad fit for our modal footer  most commonly used as a host for controls like prev/next buttons, "License Agreement"-type checkboxes or additional information about the content of the modal.

Similarly, the header has a `role="banner"`, with this definition:

> The banner role is for defining a global site header, which usually includes a logo, company name, search feature, and possibly the global navigation or a slogan. It is generally located at the top of the page.

## Describe how to reproduce the bug

1. Go to https://cookbook.kirby.design/#/examples/modal-advanced
2. Click on 'Show modal'
3. Enable 'Show Footer' in the Current modal config (at right edge of modal)
4. Open up e.g. the voiceover rotor and navigate to landmarks 
5. See that both header and footer appears as landmarks

## Add any screenshots

<img width="832" alt="Image" src="https://github.com/user-attachments/assets/cdc83afd-5fe0-4be5-a0d8-497b2910424d" />

## Please complete the following information:
- OS: [e.g. Windows, MacOS, iOS]
- Browser [e.g. Chrome, Safari]
- Browser version [e.g. 22]

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/develop/.github/SUPPORT.md#team-kirby); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.
